backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceHeaderImplBuilder.java

```diff
@@ -783,6 +786,23 @@ else if (yearId != Harvesting_Year_ID)
 		}
 	}
 
+	public int getModularContractSettingsId()
+	{
+		return MODCNTR_Settings_ID;
+	}
+
+	public void setModularContractSettingsId(final int modularContractSettingsId)
+	{
+		if (MODCNTR_Settings_ID == REPO_ID_UNSET_VALUE)
+		{
+			MODCNTR_Settings_ID = modularContractSettingsId;
+		}
+		else if (modularContractSettingsId != MODCNTR_Settings_ID)
+		{
+			M_Warehouse_ID = -1;
```

Comment:
MODCNTR_Settings_ID and consider using `de.metas.invoicecandidate.api.impl.InvoiceHeaderImplBuilder#checkOverrideID`